### ISSUE TITLE
ImageNet Classification with Deep Convolutional Neural Networks 

### ISSUE THREAD

**Issue by jskim19** [GDSC-Ewha-4th/Study-paperReview#8]
**1 Introduction**
사물 인식의 성능을 높이기 위해서는 더 큰 dataset, powerfule한 model, 그리고 overfitting을 막기 위한 더 나은 기술들이 필요
이전까지는 레이블된 이미지의 dataset이 수 만개 수준으로 작았고, 단순한 인식 작업까지는 가능
현실적인 상황에서 변동성을 고려한다면 더 큰 학습 set을 사용할 필요가 있음 
새로운 dataset은 LabelMe와 ImageNet을 포함

이 논문의 구체적 기여
1. 가장 큰 CNN을 학습시킴
2. 여태까지 중 제일 좋은 결과
3. 성능과 학습 시간 감소
4. Overfitting 방지 위한 여러 기술을 사용
5. layer 중 하나를 제거한다고 성능이 나빠지지 않음

**2 The Dataset** 

ImageNet은 22000개의 카테고리로 나눠진, 1500만장 이상의 레이블된 고해상도 이미지 dataset
Web에서 수집되고 아마존의 Mechanical Turk crowd-sourcing tool을 이용하여 레이블됨

ImageNet은 다양한 해상도의 이미지로 구성이 되어있지만 일정한 입력 차원이 필요
따라서 256 X 256으로 다운샘플링, 사전 공정 없이 raw RGB value로 학습

1. 주어진 직사각형 이미지의 짧은 변의 길이가 256이 되게 조정
2. 중앙의 256 X 256 패치 제거 

**3.1 ReLU Nonlinearity**
뉴런 출력의 표준은 tanh, sigmoid 함수를 사용
 gradient descent의 학습 시간의 관점에서 보면  non-saturation 비선형보다 느림
따라서 ReLU(Rectified Linear Units) 사용 

전통적 뉴런 모델을 대체하려는 것이 처음은 아니지만, 이는 오버피팅을 예방하기 위한 것으로 그 목적이 달랐음. 학습이 빠르면 큰 데이터셋에 학습된 큰 모델에서 좋은 결과를 보일 수 있음

**3.2 Training on Multiple GPUs**
GPU의 메모리 용량으로 인해 학습될 수 있는 네트워크 크기에 제한이 있음 
따라서 두개의 GPU에 네트워크를 나눠 사용
Parallelization scheme: 커널을 반씩 나눠 각각의 GPU에 넣음
두 개의 GPU net이 하나일 때보다 더 적은 학습시간을 필요로 함 

**3.3 Local Response Normalization** 
ReLUs는 saturating 을 막기 위해 input normalization을 할 필요가 없음
하지만 local normalization scheme은 generalization에 도움이 됨 
Without normalization 13%, with normalization 11%의 test error rate을 보임. 

**3.4 Overlapping Pooling** 
CNN의 pooling layer : 같은 커널 map에 있는 뉴런의 이웃 그룹의 출력을 요약
기존에는 overlap 하지 않았지만 이 논문에서는 s=2, z=3으로 오버래핑
결과 : 오버피팅이 덜 발생함. 

**3.5 Overall Architecture**
1st convolution layer : input size (224X224X3), 96 kernels (11X11X 3), stride=4 
2nd convolution layer : response-normalized, pooling 한 output, 256 kernels (5X5X48)
3rd convolution layer : response-normalized, pooling 한 output, 384 kernels (3X3X256)
4th convolution layer : 384 kernels (3X3X192)
5th convolution layer : 256 kernels (3X3X192)
Fully-connected layer: 4096 neurons each

**4.1 Data Augmentation** 
가장 쉽고 흔한 method : label을 유지하며 인공적으로 데이터셋을 증가
오리지널 이미지에서 적은 계산만 이루어지기 때문에 저장될 필요가 없음. 
GPU가 학습하는 동안 CPU에서 생성되기 때문에 computationally free

**4.2 Dropout**
다양한 모델에서의 예측을 합치는 것은 test error를 줄이기 위한 좋은 방법 / 큰 뉴럴 네트워크에 적용하기에는 가격이 높음 / Dropout은 두 개의 factor에 대해서만 비용이 드는 효율적인 방법
각 은닉층의 neuron의 출력을 0.5의 확률로 0으로 세팅
뉴런들은 forward pass와 back propagation에도 기여하지 않음
뉴런이 다른 뉴런의 존재에 의존할 수 없으므로 뉴런의 공동 적응을 감소 
더욱 강직한 feature만 배우게 할 수 있음
이 논문에서는 처음 두 개의 fully-connected layer에서만 dropout을 사용

**Comment by wrryu09** [comment]
잘 읽었습니다~!! 현재 상황이 어떤지, 제약이 무엇인지, 어떤 해결 방법을 썼는지 알 수 있었어요. 논문을 요약하면서 잘 모르는 용어들이 많았는데, 잘 정리해 주셔서 검색해 보면서 내용과 관련지어 더 잘 이해할 수 있을 것 같아요!!
